plugins/browser-plugin-web-vitals/src/index.ts

```diff
@@ -0,0 +1,120 @@
+
+const _trackers: Record<string, BrowserTracker> = {};
+
+declare global {
+  interface Window {
+    webVitals: {
+      onCLS(cb: ReportCallback): void;
+      onFID(cb: ReportCallback): void;
+      onLCP(cb: ReportCallback): void;
+      onINP(cb: ReportCallback): void;
+      onTTFB(cb: ReportCallback): void;
+    };
+  }
+}
+
+const WEB_VITALS_SOURCE = 'https://unpkg.com/web-vitals@3/dist/web-vitals.iife.js';
+
+interface WebVitalsPluginOptions {
+  loadWebVitalsScript?: boolean;
+  webVitalsSource?: string;
+}
+
+const defaultPluginOptions = {
+  loadWebVitalsScript: true,
+  webVitalsSource: WEB_VITALS_SOURCE,
+};
+
+/* Adds the web vitals measurements on the object used by the tracker. */
+function webVitalsListener(webVitalsObject: Record<string, unknown>) {
+  function addWebVitalsMeasurement(metricSchemaName: string): ReportCallback {
+    return (arg) => {
+      webVitalsObject[metricSchemaName] = arg.value;
+      webVitalsObject.navigationType = arg.navigationType;
+    };
+  }
+  window.webVitals.onCLS(addWebVitalsMeasurement('cls'));
+  window.webVitals.onFID(addWebVitalsMeasurement('fid'));
+  window.webVitals.onLCP(addWebVitalsMeasurement('lcp'));
+  window.webVitals.onINP(addWebVitalsMeasurement('inp'));
+  window.webVitals.onTTFB(addWebVitalsMeasurement('ttfb'));
+}
+
+/**
+ * Adds Web Vitals measurement events
+ *
+ * @param pluginOptions.loadWebVitalsScript - Should the plugin immediately load the Core Web Vitals measurement script from UNPKG CDN.
+ * @param pluginOptions.webVitalsSource - The URL endpoint the Web Vitals script should be loaded from.
+ * @remarks
+ */
+export function WebVitalsPlugin(pluginOptions: WebVitalsPluginOptions = defaultPluginOptions): BrowserPlugin {
+  const webVitalsObject: Record<string, unknown> = {};
+  const options = { ...defaultPluginOptions, ...pluginOptions };
+  let trackerId: string;
+  return {
+    activateBrowserPlugin: (tracker) => {
+      trackerId = tracker.id;
+      _trackers[trackerId] = tracker;
+
+      if (options.loadWebVitalsScript) {
+        const webVitalsScript = document.createElement('script');
+        webVitalsScript.setAttribute('src', options.webVitalsSource);
+        webVitalsScript.setAttribute('async', '1');
+        addEventListener(
+          webVitalsScript,
+          'error',
+          () => {
+            LOG.warn(`Failed to load ${options.webVitalsSource}`);
+          },
+          true
+        );
+
+        addEventListener(webVitalsScript, 'load', () => webVitalsListener(webVitalsObject), true);
+
+        document.head.appendChild(webVitalsScript);
+      } else {
+        webVitalsListener(webVitalsObject);
+      }
+
+      function sendValues() {
+        dispatchToTrackersInCollection(Object.keys(_trackers), _trackers, (t) => {
+          t.core.track(
+            buildSelfDescribingEvent({
+              event: {
+                schema: WEB_VITALS_SCHEMA,
+                data: webVitalsObject,
```

Comment:
Should we double check here that `webVitalsObject` actually has some properties set worth reporting?

Reply:
Added